### DISCUSSION TITLE
Vercel response size limitation

### DISCUSSION THREAD

**Discussion by azar-altudo** [vercel/vercel#11725] [Help]
### Question

Hi, we are creating an XML feed for all news on the website. We have thousands of news to return in the feed, ending in a 12MB response. Since Vercel has a hard limit of 4.5 MB are there any suggestions how to overcome this limit?

Feed is dynamic based on query parameters. Following suggestions are not relevant:
- Streaming, it's also failing as soon as stream size reaches 4.5 MB,
- Uploading as a static file to DAM, not valid because it should be millions of variants based on query parameter.
- Pagination is not an option since we are feeding this XML to 3rd party and it's not capable of paginating.
- Creating XML files as static file during build is not an option because we could have million of options as well as data can be changes from CMS, XML should be up to date all the time.
- We already tried to clean XML, minimize, compress and with all of this it's minimum size is 12 MB.

**Comment by versecafe** [comment]
@azar-altudo At the scale you're working at where 12MB the issue is the underlying Lambda is 6MB cap so really any AWS Lambda based solution won't work, CF workers don't actually limit response size, so if you can fit in their compute max you're perfectly fine and could spin up a simple Hono API on CF https://developers.cloudflare.com/workers/platform/limits/#response-limits. If you really want to stay on AWS you can always spin up an EC2 instance on something like tg4 for around 3-4USD a month.
